### PR TITLE
Event Queue pattern

### PR DESCRIPTION
**Pull request by qpi**
The Event Queue is one of the Game Programming Patterns, it provides an asynchronous communications protocol, meaning that the sender and receiver of the event/message do not need to interact with the queue at the same time. I implemented in this PR a sound playing queue using this pattern.

### CODE REVIEW

**Comment by qpi** [comment]
Hello @iluwatar please help me. On my PC the tests run correctly but on the Travis-CI they are failed. It says:

> No line matching interface Clip supporting format PCM_SIGNED unknown sample rate, 16 bit, stereo, 4 bytes/frame, big-endian is supported.


Originally I used WAV files, but they are not big-endian as the Travis-CI requires, so I converted them to AIF files (PCM_SIGNED 16bit stereo big-endian) but with them the tests fails too on Travis-CI
Does anyone has any idea to solve this problem?

**Comment by qpi** [comment]
Hello @iluwatar, do You receive the same error message if You change the extesion of the file to .wav or reverse to .aif? This seems strange behaviour, because on my PC(on Windows) all the test runs. Do You use Linux/OSX/Windows?

**Comment by iluwatar** [comment]
I'm on Linux. Which file should I rename? How to reverse to .aif?

**Comment by qpi** [comment]
I read about sound handling with java on linux system and it is not too stable (only on Windows and OSX works well) :( So I catch the exception and only log it. In this case the tests will pass

File: event-queue/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ - Difficulty Intermediate
+ - Queue
+---
+
+## Intent
+Event Queue is a good pattern if You have a limited accesibility resource (for example: 
+Audio or Database), but You need to handle all the requests that want to use that.
+It puts all the requests in a queue and process them asynchronously.
+Gives the resource for the event when it is the next in the queue and in same time
+removes it from the queue.
```
Comment (iluwatar) [inline event-queue/README.md]: Remove typos.

File: event-queue/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+![alt text](./etc/model.png "Event Queue")
+
+## Applicability
+Use the Event Queue pattern when
+
+* You have a limited accesibility resource and the asynchronous process is acceptable to reach that
```
Comment (iluwatar) [inline event-queue/README.md]: Remove typos.

File: event-queue/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+## Credits
+
+* [Mih�ly Kuprivecz - Event Queue]
```
Comment (iluwatar) [inline event-queue/README.md]: Add link.

File: event-queue/model.ucls [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    <end type="TARGET" refId="2" navigable="true"/>    
+    <display labels="true" multiplicity="true"/>  
+  </association>  
+  <classifier-display autosize="true" stereotype="true" package="true" initial-value="false" signature="true" 
+    sort-features="false" accessors="true" visibility="true">    
+    <attributes public="true" package="true" protected="true" private="true" static="true"/>    
+    <operations public="true" package="true" protected="true" private="true" static="true"/>  
+  </classifier-display>  
+  <association-display labels="true" multiplicity="true"/>
+</class-diagram>
```
Comment (iluwatar) [inline event-queue/model.ucls]: This file should be in event-queue/etc

File: event-queue/src/main/java/com/iluwatar/event/queue/PlayMessage.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * The Event Queue's queue will store the instances of this class.
+ * @author mkuprivecz
+ *
+ */
+public class PlayMessage {
+  AudioInputStream stream;
+  float volume;
+}
```
Comment (iluwatar) [inline event-queue/src/main/java/com/iluwatar/event/queue/PlayMessage.java]: Add constructor taking stream and volume. Store fields as private. Add getters if necessary.

**Comment by iluwatar** [comment]
@qpi looks good already. I added some review comments. Please address them and notify me when done.

File: event-queue/README.md (hunk 1 above)
Comment (qpi) [inline event-queue/README.md]: Sorry about that, but my english is not the best :( Please help me what is wrong. What do I have to change?

**Comment by qpi** [comment]
Hello @iluwatar, thank You :) I changed all the things what You requested except the typos. Please help me in that. thx.

**Comment by iluwatar** [comment]
@qpi the typos are minor so I will merge the pull request anyway.

**Comment by iluwatar** [comment]
@qpi thank you very much for the contribution :+1:

**Comment by iluwatar** [comment]
@all-contributors please add @qpi for code

**Pull request merged**